### DISCUSSION TITLE
Static Build

### DISCUSSION THREAD

**Discussion by fweth** [wakujs/waku#683] [General]
I'm just confused, there are build options `--with-*-static` for various providers, but why is there no simple `--static` option to deploy on any simple web server that can host HTML/CSS/JS?

**Comment by dai-shi** [comment]
"Static" is hosting server dependent. For example, if we were to use GitHub pages, we'd need a specific handler for it.
Which hosting service, would you like to use?

**Comment by fweth** [comment]
I just want a dist folder with an `index.html` which I can manually transfer to an old-school web server via e.g. SFTP.

**Reply by dai-shi** [reply]
Good. Then, what you want to upload is `dist/public` not `dist`.
All server components must be "static". In our template, they are "static".

**Reply by fweth** [reply]
Thank you, but it doesn't render all pages as HTML, so there is e.g. no `about.html` right?

**Reply by dai-shi** [reply]
By default, SSR is enabled, so at the build time, `dist/public/about.html` should be generated. (We could disable SSR, but it's not documented.)

**Reply by fweth** [reply]
I see, thank you!
